cognite/powerops/client/shop/cogshop_api.py

```diff
@@ -36,11 +55,364 @@ def auth(r: requests.PreparedRequest) -> requests.PreparedRequest:
+        scenario_name: str,
+        model_external_id: str | None = None,
+        scenario_external_id: str | None = None,
+        case_external_id: str | None = None,
+    ) -> ShopCaseWrite:
+        """
+        Prepare a SHOP case that can be written to cdf.
+        External ids must be unique. If they are not provided, they will be generated.
+
+        In this case, `ShopScenario` as and its `ShopModel` are mostly superfluous.
+        However, they are still added as nearly empty objects in order to set the SHOP version.
+
+        Args:
+            shop_file_list: List of 4-tuples and every item is expected.
+                    Assumes the order of the list if the order the files should be loaded into SHOP.
+                Format:
+                    `file_reference`: external if of file in CDF.
+                    `file_name`: Name of the file.
+                    `is_ascii`: Whether the file is in ASCII format.
+                    `labels`: Labels to be added to the fil, use "" if no labels
+
+            start_time: Start of time range SHOP is optimized over
+            end_time: End of time range SHOP is optimized over
+            shop_version: Version of SHOP to use (e.g. '16.0.2'or '15.7.0.0'), required.
+
+            scenario_name: Name of the scenario. Required, does not have to be unique
+            model_name: Name of the model. Required, does not have to be unique
+            scenario_external_id: External ID of the scenario. Optional, must be unique
+            model_external_id: External ID of the model. Optional, must be unique
+            case_external_id: External ID of the SHOP case. Optional, must be unique
+        Returns:
+            ShopCaseWrite: A SHOP case that can be written to CDF
+        """
+
+        scenario_write = self.prepare_shop_scenario(
+            shop_version=shop_version,
+            model_name=model_name,
+            scenario_name=scenario_name,
+            model_external_id=model_external_id,
+            scenario_external_id=scenario_external_id,
+        )
+
+        shop_files_write = [
+            ShopFileWrite(
+                name=file_name,
+                fileReference=file_reference,
+                isAscii=is_ascii,
+                label=label,
+                order=i + 1,  # Order is 1-indexed
+            )
+            for i, (file_reference, file_name, is_ascii, label) in enumerate(shop_file_list)
+        ]
+
+        case_write = ShopCaseWrite(
+            start_time=start_time,
+            end_time=end_time,
+            scenario=scenario_write,
+            shop_files=shop_files_write,
+            **({"external_id": case_external_id} if case_external_id else {}),
+        )
+        return case_write
+
+    def prepare_shop_case_with_existing_scenario(
+        self,
+        shop_file_list: list[tuple[str, str, bool, str]],
+        start_time: datetime.datetime,
+        end_time: datetime.datetime,
+        scenario_external_id: str,

```

Comment:
```suggestion
        return self.prepare_shop_case_with_existing_scenario(shop_file_list, start_time, end_time, scenario_write, case_external_id)

    def prepare_shop_case_with_existing_scenario(
        self,
        shop_file_list: list[tuple[str, str, bool, str]],
        start_time: datetime.datetime,
        end_time: datetime.datetime,
        scenario: str | ShopScenarioWrite,
```
(not a complete suggestion)
To save some duplicate code.